### DISCUSSION TITLE
Public holidays

### DISCUSSION THREAD

**Discussion by did-feedback** [Puzzlepart/did#991] [Ideas]
Would be great to have public holidays, the opportunity to turn them on and off.

**Comment by olemp** [comment]
In order to support this we need a reliable API or a manually updated table @damsleth @okms

**Reply by trondoe** [reply]
https://webapi.no/api/v1/holidays/2021

**Reply by olemp** [reply]
> https://webapi.no/api/v1/holidays/2021

Yes, seen this. We just need to find out how reliable it is.

Also, from the webpage:

> Dersom du bomber api'et med forespørsler så vil du bli stengt ute av tjenesten. Mellomlagre data lokalt der det er fornuftig. Dersom du vil ha ut mye data, så ta kontakt (stian(at)stian.net) i stedet for å skrape api'et for å ta ut alt.

**Comment by damsleth** [comment]
How about embedding the data ourselves? The data set is pretty static, and could be stored e.g. as a collection under the main-database, next to `subscriptions` and `api_tokens`:

![image](https://user-images.githubusercontent.com/7300548/138453721-36af0f97-4dfb-437d-bceb-484f5d7eeb3d.png)
 
Either `datasets`, which enables us to have (unspecified) smaller datasets stored in tables underneath, e.g. `holidays_no` for public holidays in norway,  
or a collection `holidays`, with each supported language underneath.

norwegian holidays up until 2027 is available from stavanger kommune at 
https://opencom.no/dataset/49cb0ec9-e139-4f79-bc5e-309c4faf8396/resource/f8f57a81-a22b-4c4a-b06d-2b25ee53732f

**Comment by olemp** [comment]
This is implemented.
